modules/pool/lib/staking/gauge-staking.service.ts

```diff
@@ -180,40 +190,84 @@ export class GaugeStakingService implements PoolStakingService {
         await prismaBulkExecuteOperations(operations, true, undefined);
     }
 
-    async getChildChainGaugeInfo(gaugeAddresses: string[]): Promise<{ [gaugeAddress: string]: ChildChainInfo }> {
+    /**
+     * Get the inflation rate for all the gauges on child chains.
+     * 
+     * @param gaugeAddresses
+     * @returns
+     */
+    private async getChildChainGaugeInfo(gaugeAddresses: string[]): Promise<{ [gaugeAddress: string]: GaugeRate }> {
         const currentWeek = Math.floor(Date.now() / 1000 / 604800);
-        const childChainAbi =
-            networkContext.chain === 'MAINNET'
-                ? 'function inflation_rate() view returns (uint256)'
-                : 'function inflation_rate(uint256 week) view returns (uint256)';
-        const multicall = new Multicaller3([childChainAbi]);
-
-        let response: { [gaugeAddress: string]: ChildChainInfo } = {};
+        const childChainAbi = [
+            'function inflation_rate(uint256 week) view returns (uint256)',
+            'function working_supply() view returns (uint256)'
+        ];
+        const multicall = new Multicaller3(childChainAbi);
 
         gaugeAddresses.forEach((address) => {
-            // Only L2 gauges have the inflation_rate with a week parameter
-            if (networkContext.chain === 'MAINNET') {
-                multicall.call(address, address, 'inflation_rate', [], true);
-            } else {
-                multicall.call(address, address, 'inflation_rate', [currentWeek], true);
-            }
+            multicall.call(`${address}.inflationRate`, address, 'inflation_rate', [currentWeek], true);
+            multicall.call(`${address}.workingSupply`, address, 'working_supply', [], true);
         });
 
-        const childChainData = (await multicall.execute()) as Record<string, string | undefined>;
-
-        for (const childChainGauge in childChainData) {
-            if (childChainData[childChainGauge]) {
-                response[childChainGauge] = {
-                    version: 2,
-                    rate: formatUnits(childChainData[childChainGauge]!, 18),
-                };
-            } else {
-                response[childChainGauge] = {
-                    version: 1,
-                    rate: '0.0',
-                };
+        const results = (await multicall.execute()) as {
+            [address: string]: {
+                inflationRate: BigNumber | undefined,
+                workingSupply: BigNumber
```

Comment:
v1 gauges don't have a workingSupply

Reply:
true, it will be used only in V2 and mainnet.